Rewrite execute async profiler script

ChrisKujawa opened the pull request:
## Description

 * The executeProfiling and other scripts were completely outdated
 * This PR reduces the number of scripts from three to one
 * Furthermore, rewrite executeProfiling script and make it easier to use, allowing at the end also to directly copy the flamegraph locally

Context https://camunda.slack.com/archives/C0807665N8G/p1751646838055649
<!-- Describe the goal and purpose of this PR. -->

ChrisKujawa commented on zeebe/benchmarks/docs/scripts/executeProfiling.sh:
```diff
@@ -1,18 +1,24 @@
-#   kubectl cp zeebe-0:/tmp/profiler/flamegraph-2019-03-27_12-42-33.svg .
+#   ./executeProfiling.sh <POD-NAME>
 set -oxe pipefail
 
```
Thanks 👍🏼

ChrisKujawa commented:
> @ChrisKujawa Yes i'll check this today. I'm lacking a bit of context here but to test the actual script ( also for knowledge ) will this work on any SaaS/SM cluster or are there any prerequisites?

Should work for our benchmark set ups.

panagiotisgts reviewed (approved):
LGTM! Two minor suggestions:
- Add a note on the README that this only targets `camunda/camunda` image profiling since the directory is not the same
  - Or make the script change directory if targeting a `camunda/zeebe` pod
- Delete the file from Zeebe's data dir if the copy is successful?

ChrisKujawa commented:
> LGTM! Two minor suggestions:
> 
> * Add a note on the README that this only targets `camunda/camunda` image profiling since the directory is not the same
>   
>   * Or make the script change directory if targeting a `camunda/zeebe` pod
> * Delete the file from Zeebe's data dir if the copy is successful?

Thanks @panagiotisgts good suggestion!

I added at the end a new stage to clean up local and remote. Additionally I added some checks to avoid unncessary downloads and copying. This can be interesting when rerun multiple times the profiling, then the removing part can commented out. :)

panagiotisgts commented:
@ChrisKujawa  Regarding `camunda/zeebe` deployments don't we care about them? :sweat_smile:  I guess it could be useful to have for older clusters
